### ISSUE TITLE
Syncing triggers returns InternalServerError

### ISSUE THREAD

**Issue by ZachTB123** [Azure/azure-functions-host#7156]
I'm currently trying to call sync triggers on my Function App:

```
POST https://management.azure.com/subscriptions/<my-sub>/resourceGroups/<my-resource-group>/providers/Microsoft.Web/sites/<my-function-app>/syncfunctiontriggers?api-version=2016-08-01
```

but the request is failing with:

```
Encountered an error (InternalServerError) from host runtime.
```

An example correlation id of a failure is: 3188c8d5-d20f-4e0c-b3cb-dbec803e465b.

I have multiple of these Function Apps configured the same way and this is the only one that is currently failing to sync triggers.

I have a support ticket opened (2102130040000012) but I am opening here for more visibility.

#### Investigative information

Get it from the ticket

**Comment by v-bbalaiagar** [comment]
Hi @ZachTB123, As a workaround you can delete the proxies.json file in your project (assuming you don't have any proxies).
Also, Could you please go through [azure-functions-host/issues/3978]( https://github.com/Azure/azure-functions-host/issues/3978 ) and https://github.com/microsoft/vscode-azuretools/pull/372 for more information.

**Comment by ZachTB123** [comment]
I've updated the support engineer with my findings but I'll note them here also.

I’ve been able to fix this though I would still like to know the underlying issue that caused this for future reference. What ended up working for me was:
 
1. Stop the Function App
2. Delete all the host.json and host.snapshot.<date>.json files in the azure-webjob-secrets container for my Function App in the Storage Account
3. Start the Function App
 
After deleting those files, my sync trigger requests started to work again.

Looking through the activity logs for the Function App, it seems the last successful sync trigger before this started to happen was 2021-02-12T00:39:15.9258871Z (correlation id: 334f4258-6f9d-4c88-9359-f043443ca00e). Then I start seeing failures starting from 2021-02-12T07:09:57.0186251Z (correlation id: a36334a0-a1e7-4c28-a5ea-84b59551a473). Looking through our deployment logs, it looks like the Function App ended up getting recreated between these two times. Could this have caused some sort of issues which lead to those files I deleted getting out-of-sync? Note that my Function App was still triggering and executing fine, it's just that the sync trigger requests were failing.

**Comment by v-bbalaiagar** [comment]
Hi @ZachTB123, This requires further investigation to understand the underlying issue, moving this to Triage

**Comment by ZachTB123** [comment]
I ended up closing the support ticket I had opened for this since I was able to work around it. Looking through the activity logs for the Function App, I was noticing the Function App was getting recreated every morning. The timing of this lined up with our morning redeploys. I was able to track this down to a bug in our Terraform configuration since that was mistakingly recreating the Function App. I don't have a solid understanding of what caused the original issue, but my thought was that this constant recreation was causing those host snapshot files to be created which somehow got everything into a bad state since deleting all the files and then restarting the Function App fixed it.
